Upstream-Status: Submitted [https://crrev.com/c/1213180]

- Submitted to upstream, waiting approval

This patch fixes such errors as:

../../third_party/blink/renderer/platform/scroll/scroll_snap_data.h:22:48:
error: changes meaning of 'SnapContainerData' from 'using
SnapContainerData = class cc::SnapContainerData' [-fpermissive]
using SnapContainerData = cc::SnapContainerData;

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 5b1c4cfa6b7f934280ac5a7b1a2d38aa6ecc05e1 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Fri, 7 Sep 2018 19:55:59 +0300
Subject: [PATCH] ScrollPaintPropertyNode: Rename SnapContainerData() to
 GetSnapContainerData()

GCC is stricter than clang when it comes to class members' names and how
they can change the meaning of a previously existing symbol with the same
name.

Here is a short error message:

error: changes meaning of 'SnapContainerData' from 'using SnapContainerData =
class cc::SnapContainerData' [-fpermissive]  using SnapContainerData =
cc::SnapContainerData;

Cq-Include-Trybots: luci.chromium.try:linux_layout_tests_slimming_paint_v2;master.tryserver.blink:linux_trusty_blink_rel
Change-Id: I0de5460f17b13484253e1ff0538b111c9a3b0d13
---
 .../renderer/platform/graphics/compositing/property_tree_manager.cc     | 2 +-
 .../blink/renderer/platform/graphics/paint/scroll_paint_property_node.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc b/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
index 2ac29aa32ae7..dda9f0a6b124 100644
--- a/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
+++ b/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
@@ -299,7 +299,7 @@ void PropertyTreeManager::CreateCompositorScrollNode(
           scroll_node->OverscrollBehaviorX()),
       static_cast<cc::OverscrollBehavior::OverscrollBehaviorType>(
           scroll_node->OverscrollBehaviorY()));
-  compositor_node.snap_container_data = scroll_node->SnapContainerData();
+  compositor_node.snap_container_data = scroll_node->GetSnapContainerData();
 
   auto compositor_element_id = scroll_node->GetCompositorElementId();
   if (compositor_element_id) {
diff --git a/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h b/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
index 2bd206ee9600..2d222bcea8e5 100644
--- a/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
+++ b/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
@@ -99,7 +99,7 @@ class PLATFORM_EXPORT ScrollPaintPropertyNode
     return state_.overscroll_behavior.y;
   }
 
-  base::Optional<SnapContainerData> SnapContainerData() const {
+  base::Optional<SnapContainerData> GetSnapContainerData() const {
     return state_.snap_container_data;
   }
 
-- 
2.11.0

